randovania/game_connection/builder/dolphin_connector_builder.py

```diff
@@ -11,12 +12,62 @@
 
 
 class DolphinConnectorBuilder(PrimeConnectorBuilder):
+    dolphin_cmd: str
+    dolphin_comm: str
+    dolphin_child: Popen
```

Comment:
feel like there could be some better ways for dolphin_cmd and dolphin_comm, will revisit that later

dolphin_child i'd rename to dolphin_child_process for clarity.

Reply:
Ideally we'd do away with `dolphin_comm` entirely, but DME is really constraining here: it looks for a process with the right memory map *and* a commname exactly matching DME_DOLPHIN_PROCESS_NAME (or `dolphin-emu` if unset).

My preference would be to simply pass it the PID to attach to. There's also room for improving the autodetection -- look at argv0 as well as commname, or even ignore the program name entirely and just look for the memory map, which is highly distinctive.

These would not be terribly complicated features, but they'd need to be added to upstream DME (or added as a patch series on the C++ parts of pyDME and rebased whenever upstream changes are pulled in).

Reply:
DME has code for supporting multiple hooked Dolphins at once and I do wish to have it exposed in pyDME. In this case you'd have the global API that works as is right now, then we could also have an API for communicating with an specific PID.

Reply:
My understanding, though, is that pyDME includes (a subset of) upstream DME to handle the actual interface with dolphin, and that not diverging from upstream too much is a priority.

Reply:
pyDME is our (specifically Henrique's) library, so I trust him when he suggests we modify its API